OpenRA.Mods.Common/Traits/Buildings/Exit.cs

```diff
@@ -77,5 +77,29 @@ public static Exit RandomExitOrDefault(this Actor actor, World world, string pro
 			var shuffled = allOfType.Shuffle(world.SharedRandom);
 			return p != null ? shuffled.FirstOrDefault(p) : shuffled.First();
 		}
+	
+		public static Exit ClosestExitOrDefault(this Actor actor, World world, string productionType, Lazy<RallyPoint> rp, Func<Exit, bool> p = null)
+		{
+			if (!rp.Value.Sorted)
```

Comment:
This is not a hot path, and the sorting is cheap, so please remove the caching here. The additional code complexity outweighs any potential performance benefit.